### PR TITLE
Remove from SSL_COMP_add_compression_method from API

### LINKED ISSUE #2615: Cannot add new compression schemes
In 1.1.0, by making ```COMP_METHOD``` and ```COMP_CTX`` we inadvertently made it impossible to add new compression schemes at run-time; you have to modify the OpenSSL source.

By our definitions, this is a bug.  This issue records that bug.

### PR DESCRIPTION
**Pull request by richsalz**
Will fail at compile-time with warnings enabled.
It always returns an error now, for backward ABI compat.
Fixed #2615

### CODE REVIEW

File: ssl/ssl_ciph.c [hunk 1]
```diff
@@ -1776,6 +1776,12 @@ SSL_COMP *ssl3_comp_find(STACK_OF(SSL_COMP) *sk, int n)
     return (NULL);
 }
 
+int SSL_COMP_add_compression_method(int id, COMP_METHOD *cm);
+int SSL_COMP_add_compression_method(int id, COMP_METHOD *cm)
+{
+    return 1;
```
Comment (vdukhovni) [inline ssl/ssl_ciph.c]: IIRC `return 0` is the correct error result.

File: util/libssl.num [hunk 2]
```diff
@@ -24,7 +24,7 @@ SSL_set_generate_session_id             24	1_1_0	EXIST::FUNCTION:
 SSL_get_ex_data_X509_STORE_CTX_idx      25	1_1_0	EXIST::FUNCTION:
 SSL_get_quiet_shutdown                  26	1_1_0	EXIST::FUNCTION:
 SSL_dane_enable                         27	1_1_0	EXIST::FUNCTION:
-SSL_COMP_add_compression_method         28	1_1_0	EXIST::FUNCTION:
+SSL_COMP_add_compression_method         28	1_1_0	NOEXIST::FUNCTION:
```
Comment (vdukhovni) [inline util/libssl.num]: We should keep the public ABI entry point.

Comment (richsalz) [inline util/libssl.num]: ok fixed.

File: ssl/ssl_ciph.c (hunk 1 above)
Comment (richsalz) [inline ssl/ssl_ciph.c]: yes.  fixed.

**Comment by richsalz** [comment]
updated commit pushed.  for matser and 1.1.0

File: doc/man3/SSL_COMP_get_id.pod [hunk 3]
```diff
@@ -23,24 +22,15 @@ Deprecated:
...
 
 SSL_COMP_get_id() returns the id of the compression method B<comp>.
 
-In versions of OpenSSL prior to 1.1.0 SSL_COMP_free_compression_methods() freed
-the internal table of compression methods that were built internally, and
-possibly augmented by adding SSL_COMP_add_compression_method(). However this is
-now unnecessary from version 1.1.0.  No explicit initialisation or
-de-initialisation is necessary. See L<OPENSSL_init_crypto(3)> and
-L<OPENSSL_init_ssl(3)>. From OpenSSL 1.1.0 calling this function does nothing.
+In versions of OpenSSL prior to 1.1.0 SSL_COMP_free_compression_methods();
```
Comment (vdukhovni) [inline doc/man3/SSL_COMP_get_id.pod]: I think some text got lost here...

**Comment by vdukhovni** [comment]
I think it still remains to move the header definition from include/openssl to a suitable internal include file.

**Comment by richsalz** [comment]
If it's not declared in a header file, then it doesn't show up in libssl.num and therefore not exposed as a public symbol.  Arguably a tooling limit, but not one I'm going to try to fix.

File: ssl/ssl_ciph.c [hunk 4]
```diff
@@ -1776,6 +1776,11 @@ SSL_COMP *ssl3_comp_find(STACK_OF(SSL_COMP) *sk, int n)
     return (NULL);
 }
 
+int SSL_COMP_add_compression_method(int id, COMP_METHOD *cm)
+{
+    return 0;
```
Comment (levitte) [inline ssl/ssl_ciph.c]: Add an error on the error stack.

**Comment by richsalz** [comment]
I originally thought this was a simple fix.  I'm going to leave the bug open and stop on this PR.  There are more important things to work on. :)

**Pull request closed without merging**